Sources/FluentKit/Properties/Parent.swift

```diff
@@ -1,207 +1,231 @@
+                try container.encode([
+                    To.Wrapped.key(for: \._$id): self.id
+                ])
+            }
         }
 
-        init(key: String) {
-            self.storage = []
-            self.key = key
+        func decode(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: _ModelCodingKey.self)
+            try self.parent._id.decode(from:
+                container.superDecoder(forKey: .string(To.Wrapped.key(for: \._$id)))
+            )
+            // TODO: allow for nested decoding
         }
 
-        func prepare(query: inout DatabaseQuery) {
-            // we can assume query.schema since eager loading
-            // is only allowed on the base schema
-            query.joins.append(.model(
-                foreign: .field(path: [To.key(for: \._$id)], schema: To.schema, alias: nil),
-                local: .field(path: [self.key], schema: query.schema, alias: nil),
-                method: .inner
-            ))
-            query.fields += To().fields.map { (_, field) in
-                return .field(
-                    path: [field.key],
-                    schema: To.schema,
-                    alias: To.schema + "_" + field.key
-                )
+        func eagerLoad(from eagerLoads: EagerLoads, label: String) throws {
+            guard let request = eagerLoads.requests[label] else {
+                return
             }
-        }
 
-        func run(models: [AnyModel], on database: Database) -> EventLoopFuture<Void> {
-            do {
-                self.storage = try models.map { child in
-                    return try child.joined(To.self)
-                }
-                return database.eventLoop.makeSucceededFuture(())
-            } catch {
-                return database.eventLoop.makeFailedFuture(error)
+            if let join = request as? JoinEagerLoad {
+                self.parent.eagerLoadedValue = try join.get(id: self.id)
+            } else if let subquery = request as? SubqueryEagerLoad {
+                self.parent.eagerLoadedValue = try subquery.get(id: self.id)
+            } else {
+                fatalError("unsupported eagerload request: \(request)")
             }
         }
 
-        func get(id: To.IDValue) throws -> To? {
-            return self.storage.filter { parent in
-                return parent.id == id
-            }.first
+        func eagerLoad(to eagerLoads: EagerLoads, method: EagerLoadMethod, label: String) {
+            switch method {
+            case .subquery:
+                eagerLoads.requests[label] = SubqueryEagerLoad(key: self.key)
+            case .join:
+                eagerLoads.requests[label] = JoinEagerLoad(key: self.key)
+            }
         }
     }
 }
+
+// MARK: - Optional Model Identifiable
+
+extension Optional: ModelIdentifiable where Wrapped: ModelIdentifiable {
+    public typealias IDValue = Wrapped.IDValue?
+
+    public var id: Wrapped.IDValue?? {
```

Comment:
Does this need to be a double optional? I think we could get away with just `Wrapped.IDValue`. But I'm also not sure we need `ModelIdentifiable` at all (see earlier comment)